core-primitives/stf-executor/src/executor_tests.rs

```diff
@@ -0,0 +1,317 @@
+use itp_types::H256;
+use sgx_externalities::SgxExternalitiesTrait;
+use sgx_tstd::panic;
+use sp_core::Pair;
+use sp_runtime::app_crypto::sp_core::blake2_256;
+use std::{sync::Arc, time::Duration, vec, vec::Vec};
+
+// FIXME: Create unit tests for update_states, execute_shield_funds, execute_trusted_call, execute_trusted_call_on_stf #554
+
+pub fn propose_state_update_executes_all_calls_given_enough_time() {
+	// given
+	let (stf_executor, ocall_api, state_handler) = stf_executor();
+	let mrenclave = ocall_api.get_mrenclave_of_self().unwrap().m;
+	let (_, shard) = init_state_and_shard_with_state_handler(state_handler.as_ref());
+	let sender = endowed_account();
+	let signed_call = TrustedCall::balance_transfer(
+		sender.public().into(),
+		sender.public().into(),
+		42,
+	)
+	.sign(&sender.clone().into(), 0, &mrenclave, &shard);
+	let call_operation_hash: H256 =
+		blake2_256(&signed_call.clone().into_trusted_operation(true).encode()).into();
+	let signed_call_two = TrustedCall::balance_transfer(
+		sender.public().into(),
+		sender.public().into(),
+		100,
+	)
+	.sign(&sender.clone().into(), 1, &mrenclave, &shard);
+	let call_operation_hash_two: H256 =
+		blake2_256(&signed_call_two.clone().into_trusted_operation(true).encode()).into();
+
+	let old_state_hash = state_hash(&state_handler.load_initialized(&shard).unwrap());
+
+	// when
+	let batch_execution_result = stf_executor
+		.propose_state_update(
+			&vec![signed_call.clone(), signed_call_two.clone()],
+			&ParentchainHeaderBuilder::default().build(),
+			&shard,
+			Duration::from_secs(1000),
+			|state| state,
+		)
+		.unwrap();
+
+	// then
+	assert_eq!(old_state_hash, batch_execution_result.state_hash_before_execution);
+	assert_eq!(batch_execution_result.executed_operations.len(), 2);
+	assert_eq!(
+		batch_execution_result.get_executed_operation_hashes(),
+		vec![call_operation_hash, call_operation_hash_two]
+	);
+	// Ensure that state has been updated and not actually written.
+	assert_ne!(
+		state_handler.load_initialized(&shard).unwrap(),
+		batch_execution_result.state_after_execution
+	);
+}
+
+pub fn propose_state_update_executes_only_one_trusted_call_given_not_enough_time() {
+	// given
+	let (stf_executor, ocall_api, state_handler) = stf_executor();
+	let mrenclave = ocall_api.get_mrenclave_of_self().unwrap().m;
+	let (_, shard) = init_state_and_shard_with_state_handler(state_handler.as_ref());
+	let sender = endowed_account();
+	let signed_call = TrustedCall::balance_transfer(
+		sender.public().into(),
+		sender.public().into(),
+		42,
+	)
+	.sign(&sender.clone().into(), 0, &mrenclave, &shard);
+	let call_operation_hash: H256 =
+		blake2_256(&signed_call.clone().into_trusted_operation(true).encode()).into();
+	let signed_call_two = TrustedCall::balance_transfer(
+		sender.public().into(),
+		sender.public().into(),
+		100,
+	)
+	.sign(&sender.clone().into(), 0, &mrenclave, &shard);
+
+	let old_state_hash = state_hash(&state_handler.load_initialized(&shard).unwrap());
+
```

Comment:
Extracting could be done now, but it will probably be changed again when solving issue https://github.com/integritee-network/worker/issues/554 as still a lot of tests are missing  (where actual state changes need to be tested for). So I think thinking about how to extract code is not worth it's time for now.